apps/authentication-portal/src/main/webapp/basicauth.jsp

```diff
@@ -77,37 +77,13 @@
                     if (usernameUserInput) {
                         var usernameUserInputValue = usernameUserInput.value.trim();
 
-                        if (tenantName && tenantName !== "null") {
-
-                            if (isEmailUsernameEnabled) {
-
-                                if (usernameUserInputValue.split("@").length <= 1) {
-                                    var errorMessage = document.getElementById("error-msg");
-
-                                    errorMessage.innerHTML = "Invalid Username. Username has to be an email address.";
-                                    errorMessage.style.display = "block";
-
-                                    return;
-                                }
-
-                                if (usernameUserInputValue.split("@").length === 2) {
-                                    userName.value = usernameUserInputValue + "@" + tenantName;
-                                }
-                                else {
-                                    userName.value = usernameUserInputValue;
-                                }
-                            } else {
-                                if (usernameUserInputValue.split("@").length > 1) {
-                                    userName.value = usernameUserInputValue;
-                                } else {
-                                    userName.value = usernameUserInputValue + "@" + tenantName;
-                                }
-
-                            }
-                            
-                        } else {
-                            userName.value = usernameUserInputValue;
+                        if (isEmailUsernameEnabled && usernameUserInputValue.split("@").length <= 1) {
```

Comment:
Shall we do this validation also from the back-end ?

Reply:
If we are going to do this validation from the back-end. We have to handle it at authentication framework/basic authenticator. In that case we can't display a proper error message in UI as InvalidCredentialException is handled at authentication framework. I think it is better to keep the validation at this level.

Reply:
Back-end validation was added by introducing a new AuthFailureMessage.